src/c2goto/library/stdlib.c

```diff
@@ -8,14 +8,43 @@
 #undef atol
 #undef getenv
 
+short atexit_index = 0;
+void (*atexit_func[32])();
+
+void __atexit_handler()
+{
+__ESBMC_HIDE:;
+  for(short i = atexit_index - 1; i >= 0; --i)
+    atexit_func[i]();
+}
+
+int atexit(void (*func)(void))
+{
+__ESBMC_HIDE:;
+  // Max of 32 functions can be registered
```

Comment:
Why do we use this magic number?

Reply:
[C99 7.20.4.2](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1124.pdf#%5B%7B%22num%22%3A651%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C-27%2C816%2Cnull%5D)

Reply:
Should be an at least, not a max.

## Description
The `atexit` function registers the function pointed to by `func`, to be called without arguments at normal program termination.

## Environmental limits
The implementation shall support the registration of at least 32 functions.

Reply:
Changed to an infinite array